MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -36,6 +36,35 @@ open class RouteProgress: NSObject {
         return route.legs[legIndex]
     }
 
+    /**
+     Returns the remaining steps left on the current route
+     */
+    @objc public var remainingSteps: [RouteStep] {
+        var steps = [RouteStep]()
+        
+        // get all remaining steps on currentLeg
+        let curStepIndex = currentLegProgress.stepIndex
+        var curLegSteps = currentLeg.steps.suffix(from: curStepIndex)
+        _ = curLegSteps.popLast()
+        
+        for step in curLegSteps {
+            steps.append(step)
+        }
+        
+        // get all remaining steps after currentLeg
+        let remainingLegs = route.legs.suffix(from: legIndex + 1)
+        for leg in remainingLegs {
+            var curLegSteps = leg.steps
+            _ = curLegSteps.popLast()
+            
+            for step in curLegSteps {
+                steps.append(step)
```

Comment:
Implement a `remainingLegs` computed property like this:

```swift
return Array(legs.suffix(from: legIndex + 1))
```

Then this method becomes simply:

```swift
return currentLegProgress.remainingSteps + remainingLegs.flatMap { $0.steps }
```

The one wrinkle is that you might want to merge waypoint arrivals with the subsequent departures. We do that in a few places, but it would be misleading to do that as part of a simply named `RouteProgress.remainingSteps` method. So the caller could do something like this:

```swift
routeProgress.remainingSteps.filter { $0.maneuverType != .depart }
```